server/src/instant/model/schema.clj

```diff
@@ -141,50 +141,171 @@
+  (str etype "->" label ": "
+       "Duplicate entry found for attribute. "
+       "Check your schema file for duplicate link definitions. "
+       "If it's not in the schema file, it may have been generated by the backend. "
+       "Check your full schema in the dashboard: "
+       "https://www.instantdb.com/dash?s=main&t=explorer"))
+
+(defn backwards-link-message [[etype label]]
+  (str etype "->" label ": "
+       "Conflicting link found for attribute. "
        "It's possible that you already have a link with the same label names, but in the reverse direction. "
-       "We cannot automatically swap the direction of the link, so you will need to do this manually. "
-       "Check your schema in https://www.instantdb.com/dash for a link with the same label names. "
-       "If you find one, you can either delete the existing link in the dashboard, or swap the `forward` and `reverse` parameters for this link in your schema file."))
+       "We cannot automatically swap the direction of the link. "
+       "To fix this, can: a) swap the `forward` and `reverse` parameters for this link in your schema file, or b) delete the existing link in the dashboard."
+       "Check your full schema in the dashboard for a link with the same label names: "
+       "https://www.instantdb.com/dash?s=main&t=explorer"))
 
 (defn assert-unique-idents! [current-attrs steps]
-  (let [current-fwd-ident-names (->> current-attrs
-                                     (map attr-model/fwd-ident-name)
-                                     (map vec))
-        current-all-ident-names (->> current-attrs
-                                     (mapcat attr-ident-names)
-                                     (map vec))
-        new-ident-names (->>
-                         steps
-                         (mapcat (fn [[op data]]
-                                   (when (= op :add-attr)
-                                     (attr-ident-names data)))))
-        dups (->> (concat current-all-ident-names new-ident-names)
-                  (frequencies)
-                  (filter (fn [[_ freq]] (> freq 1))))
-        errors (map (fn [[[etype label]]]
-                      {:in [:schema]
-                       :message (if
-                                 (some #(= % [etype label]) current-fwd-ident-names)
-                                  (dup-message etype label)
-                                  (backwards-link-message etype label))})
-                    dups)]
-    (ex/assert-valid! :schema
-                      :steps
-                      errors)))
+  (let [current-link-attrs
+        (filter (comp #{:ref} :value-type) current-attrs)
+
+        current-blobs
+        (filter (comp #{:blob} :value-type) current-attrs)
+
+        current-blob-idents
+        (->> current-blobs
+             (map #(attr-model/fwd-ident-name %))
+             (into #{}))
+
+        current-links-mapping-fwd
+        (->> current-link-attrs
+             (map #(vector
```

Comment:
If you want to get a bit clojure fancy, I _think_ you can write this as: 

(map (juxt attr-model/fwd-ident-name attr-model/rev-ident-name